### PR TITLE
Change some custom collections to inherit from List<T> to avoid dupli…

### PR DESCRIPTION
**Pull request by LorenVS**
Remove a bunch of custom code that is already provided by List<T>. Keep custom types for the Editor attributes.

### CODE REVIEW

File: src/managed/OpenLiveWriter.ApplicationFramework/MenuDefinitionEntryCollection.cs [hunk 1]
```diff
@@ -12,233 +13,28 @@ namespace OpenLiveWriter.ApplicationFramework
...
-            /// <summary>
-            /// Sets the enumerator to its initial position, which is before the first element in the collection.
-            /// </summary>
-            void IEnumerator.Reset()
-            {
-                baseEnumerator.Reset();
-            }
+            Add(mde);
         }
+
```
Comment (KirillOsenkov) [inline src/managed/OpenLiveWriter.ApplicationFramework/MenuDefinitionEntryCollection.cs]: We might want to support StyleCop at some point, and this empty line violates (http://stylecop.com/docs/SA1508.html)[http://stylecop.com/docs/SA1508.html]

**Comment by kathweaver** [comment]
Nothing appears to have been broken.

**Comment by KirillOsenkov** [comment]
Yes, the change is fine. Ignore my previous comment.

**Comment by RobDolinMS** [comment]
Thanks @kathweaver and @KirillOsenkov for reviewing.  Merging...

**Pull request merged**